test/parallel/test-child-process-flush-stdio.js

```diff
@@ -20,10 +21,42 @@ function spawnWithReadable() {
     assert.strictEqual(code, 0);
     assert.strictEqual(signal, null);
     assert.strictEqual(Buffer.concat(buffer).toString().trim(), '123');
+    spawnWithPipe();
   }));
   p.stdout.on('readable', function() {
-    let buf;
-    while (buf = this.read())
-      buffer.push(buf);
+    setImmediate(() => {
+      let buf;
+      while (buf = this.read())
+        buffer.push(buf);
+    });
   });
 }
+
+function spawnWithPipe() {
+  const buffer = [];
+  const through = new stream.PassThrough();
+  const p = cp.spawn('seq', [ '36000' ]);
+
+  const ret = [];
+  for (let i = 1; i <= 36000; i++) {
+    ret.push(i);
+  }
+
+  p.on('close', common.mustCall(function(code, signal) {
+    assert.strictEqual(code, 0);
+    assert.strictEqual(signal, null);
+    assert.strictEqual(Buffer.concat(buffer).toString().trim(), ret.join('\n'));
+  }));
+
+  p.on('exit', common.mustCall(function() {
```

Comment:
This should read `close`, as there’s currently no guarantee that `'close'` will be called before `'exit'`.